test/e2e/test-cmd.sh

```diff
@@ -104,34 +92,33 @@ kubectl version
 # This is to enable docker push. Running it here to fail early.
 gcloud auth configure-docker
 
-# TODO: create more objects.
-# STEP 1: create an object
-
-# We use controller revision as the test subject because it has multiple
-# versions, and no controller is going to write to it.
-cat <<EOF | kubectl create --namespace=default -f -
-apiVersion: apps/v1
-kind: ControllerRevision
-metadata:
-  name: sample
-data:
-  Raw: a
-revision: 1
-EOF
+# TODO: Test more types of resources.
 
-kubectl get controllerrevisions sample --namespace=default
+# STEP 1: create an object encoded in a non-default storage version. We cannot
+# create the object via the apiserver, because apiserver always encode the
+# object to the default storage version before storing in etcd.
 
-# STEP 2: change the apiserver configuration of --storage-versions.
+# Copy the pre-made proto file of the object to the master machine.
+user_name=$(gcloud compute --project "${PROJECT}" ssh --zone "${KUBE_GCE_ZONE}" "${CLUSTER_NAME}-master" --command "whoami")
+gcloud compute scp "${MIGRATORROOT}/test/e2e/${TESTFILE}" "${user_name}@${CLUSTER_NAME}-master:~/" --project "${PROJECT}" --zone "${KUBE_GCE_ZONE}"
 
-# Change the apiserver manifest to change the storage version.
-# TODO: Set the storage version based on the version of the apiserver. There is
-# no guarantee that apps/v1beta2 is always supported.
+# Get the etcd container ID.
+result=$(gcloud compute --project "${PROJECT}" ssh --zone "${KUBE_GCE_ZONE}" "${CLUSTER_NAME}-master" --command \
+  "docker ps")
+etcd_container=$(echo "${result}" | grep "etcd-server-${CLUSTER_NAME}-master" | grep -v pause | cut -d ' ' -f 1)
+
+# Copy the proto file to the etcd container
+gcloud compute --project "${PROJECT}" ssh --zone "${KUBE_GCE_ZONE}" "${CLUSTER_NAME}-master" --command \
+  "docker cp ${TESTFILE} ${etcd_container}:/"
+
+# Create the object via etcdctl
 gcloud compute --project "${PROJECT}" ssh --zone "${KUBE_GCE_ZONE}" "${CLUSTER_NAME}-master" --command \
-"sudo sed -i \"s/--v=/--storage-versions=apps\/v1beta2 --v=/\" /etc/kubernetes/manifests/kube-apiserver.manifest"
+  "docker exec ${etcd_container} /bin/sh -c \"cat /${TESTFILE} | ETCDCTL_API=3 etcdctl put /registry/controllerrevisions/default/sample\""
 
-wait-for-apiserver 
+# Validate that the creation via etcdctl is successful
+kubectl get controllerrevisions sample --namespace=default
```

Comment:
better to verify that version is v1beta2 (using the same steps below that verifies apps/v1)